cpp/tests/test_graph_simulation.cpp

```diff
@@ -237,9 +237,9 @@ TEST(TestGraphSimulation, consistencyStochasticMobility)
     auto result_n0 = sim.get_graph().nodes()[0].property.get_result().get_last_value();
     auto result_n1 = sim.get_graph().nodes()[1].property.get_result().get_last_value();
 
-    auto expected_values_n0 = std::vector<double>{692.0, 43.630772796677256, 95.750528156188381, 159.61869904713436};
+    auto expected_values_n0 = std::vector<double>{692.0, 43.647065696491225, 95.758901343443526, 159.59403296006528};
     auto actual_values_n0   = std::vector<double>{result_n0[0], result_n0[1], result_n0[2], result_n0[3]};
-    auto expected_values_n1 = std::vector<double>{708.0, 44.063147085799322, 96.485223892060375, 160.45162902214025};
+    auto expected_values_n1 = std::vector<double>{708.0, 44.04685402850226, 96.476851270558342, 160.47629470093941};
```

Comment:
Can you explain to me why these values have to be changed?

Reply:
Yes, it is due to the newly forced steps. Disabling forced step sizing (i.e. overriding the bool with true) does recover the old results.
The graph simulation makes small time steps in which it calls each node simulation, which usually calls advance with `dt < t_max - t`, hence it makes only one forced time step. There are at least two steps smaller than dt_min, hence the discrepancy.

Reply:
Actually, there was a problem with the OdeIntegrator. In its advance function, I forced a step whenever making the last step, i.e. `dt < t_max - t`. However, we do not care whether this is the last step or not, we want to know whether the last step would be smaller than dt_min (so whenever the step with `dt = t_max - t` would fail to adapt). Only then we actually need to force a smaller step, otherwise we may loose accuracy.

The new logic in advance re-takes a step, which is not optimal, but currently there is no way to get dt_min from advance. Maybe we need another virtual method?